packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -302,12 +302,12 @@ DevFSContent? processSkSLBundle(String? bundlePath, {
     throw Exception('SkSL bundle was invalid');
   }
 
-  final String? parsedPlatform = bundle['platform'] as String?;
-  TargetPlatform? bundleTargetPlatform;
-  if (parsedPlatform != null) {
-    bundleTargetPlatform = getTargetPlatformForName(parsedPlatform);
-  }
-  if (bundleTargetPlatform == null || bundleTargetPlatform != targetPlatform) {
+  final TargetPlatform? bundleTargetPlatform = switch (bundle) {
```

Comment:
I believe this introduces a change in behavior (`bundle['platform']` could possibly be something that is both non-null and not a `String`.

```suggestion
  final TargetPlatform? bundleTargetPlatform = switch (bundle as String?) {
```